src/Microsoft.AspNet.Mvc.Razor.Host/MvcRazorParser.cs

```diff
@@ -58,43 +61,63 @@ public class MvcRazorParser : RazorParser
             var descriptors = new List<TagHelperDirectiveDescriptor>();
 
             // For tag helpers, the @removeTagHelper only applies tag helpers that were added prior to it.
-            // Consequently we must visit tag helpers outside-in - furthest _ViewStart first and nearest one last. This
-            // is different from the behavior of chunk merging where we visit the nearest one first and ignore chunks
-            // that were previously visited.
-            var chunksFromViewStarts = inheritedCodeTrees.Reverse()
-                                                         .SelectMany(tree => tree.Chunks);
-            var chunksInOrder = defaultInheritedChunks.Concat(chunksFromViewStarts);
+            // Consequently we must visit tag helpers outside-in - furthest _GlobalImport first and nearest one last.
+            // This is different from the behavior of chunk merging where we visit the nearest one first and ignore
+            // chunks that were previously visited.
+            var chunksFromGlobalImports = inheritedCodeTrees
+                .Reverse()
+                .SelectMany(tree => tree.Chunks);
+            var chunksInOrder = defaultInheritedChunks.Concat(chunksFromGlobalImports);
             foreach (var chunk in chunksInOrder)
             {
-                var addHelperChunk = chunk as AddTagHelperChunk;
-                if (addHelperChunk != null)
+                // All TagHelperDirectiveDescriptors created here have undefined source locations because the source 
+                // that created them is not in the same file.
```

Comment:
side question: does the lack of filename in `SourceLocation` lead to any observable problems?  for example do we find any errors after parsing global import or layout files i.e. when back in the view?  wondering if this comment points to a minor quirk or a real issue.